packages/ui/src/components/layout-v2.tsx

```diff
@@ -0,0 +1,182 @@
+import { FC, HTMLAttributes, ReactNode } from 'react'
+
+import { cn } from '@utils/cn'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+interface LayoutProps {
+  children?: ReactNode
+  className?: string
+  gap?: string
+  gapX?: string
+  gapY?: string
+  as?: 'div' | 'span'
+}
+
+interface FlexProps extends LayoutProps, VariantProps<typeof flexVariants> {}
+
+const gridVariants = cva('grid', {
+  variants: {
+    align: {
+      start: 'items-start',
+      center: 'items-center',
+      end: 'items-end',
+      baseline: 'items-baseline',
+      stretch: 'items-stretch'
+    },
+    justify: {
+      start: 'justify-start',
+      center: 'justify-center',
+      end: 'justify-end',
+      between: 'justify-between'
+    }
+  }
+})
+
+interface GridProps extends LayoutProps, VariantProps<typeof gridVariants> {
+  columns?: string
+  rows?: string
+  flow?: 'row' | 'column' | 'dense' | 'row-dense' | 'column-dense'
+}
+
+const flexVariants = cva('flex', {
+  variants: {
+    direction: {
+      row: 'flex-row',
+      column: 'flex-col',
+      'row-reverse': 'flex-row-reverse',
+      'column-reverse': 'flex-col-reverse'
+    },
+    align: {
+      start: 'items-start',
+      center: 'items-center',
+      end: 'items-end',
+      baseline: 'items-baseline',
+      stretch: 'items-stretch'
+    },
+    justify: {
+      start: 'justify-start',
+      center: 'justify-center',
+      end: 'justify-end',
+      between: 'justify-between'
+    },
+    wrap: {
+      nowrap: 'flex-nowrap',
+      wrap: 'flex-wrap',
+      'wrap-reverse': 'flex-wrap-reverse'
+    }
+  },
+  defaultVariants: {
+    direction: 'row'
+  }
+})
+
+const Flex = ({
+  children,
+  className,
+  direction,
+  align,
+  justify,
+  gap,
+  gapX,
+  gapY,
+  wrap,
+  as: Comp = 'div',
+  ...props
+}: FlexProps & HTMLAttributes<HTMLDivElement>) => {
+  return (
+    <Comp
+      className={cn(
+        flexVariants({ direction, align, justify, wrap }),
+        gap && `gap-${gap}`,
```

Comment:
instead of allowing all possible values, I think we should limit `gap` to a few predefined values, controlled via design tokens like `sm`, `md`, `lg` etc.